app/src/main/java/com/example/android_level_3/viewmodel/MainViewModel.kt

```diff
@@ -1,96 +1,179 @@
+import com.example.android_level_3.old_classes.TestContact
+import com.example.android_level_3.retrofit.Retrofit
+import com.example.android_level_3.retrofit.RetrofitServerApi
+import com.example.android_level_3.retrofit.model.Contact
+import com.example.android_level_3.retrofit.model.ContactId
+import com.example.android_level_3.retrofit.model.ContactsServerResponse
+import com.example.android_level_3.retrofit.model.CreateUserModel
+import com.example.android_level_3.retrofit.model.ExtensionServerResponse
+import com.example.android_level_3.retrofit.model.ServerResponse
+import com.example.android_level_3.retrofit.model.UserAuthorisationEntity
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withTimeoutOrNull
+import retrofit2.Response
+import java.net.ConnectException
 
 open class MainViewModel: ViewModel() {
 
-    private val contactList = MutableLiveData<List<Contact>>()
-    val observableContactList: LiveData<List<Contact>> = contactList
+    private val contactList = MutableLiveData<List<TestContact>>()
+    val observableContactList: LiveData<List<TestContact>> = contactList
+
+    val tabLayoutVisibility = MutableLiveData<Boolean>(true)
+
+    var serverApi: RetrofitServerApi
+
+    // данные юзера необходимые для дальнейших запросов
+    val token = MutableLiveData<String>()
+    val userId = MutableLiveData<Int>()
 
-    private var deletedContactData: Contact? = null
-    private var deletedContactPosition: Int = 0
-    private var timer: CountDownTimer? = null
+    // поля состояния видимости для режима поиска (список всех юзеров, список контактов)
+    val isActiveSearchAddContact = MutableLiveData<Boolean>(false)
+    val isActiveSearchUserContacts = MutableLiveData<Boolean>(false)
 
-    val onTickTimerMessage = MutableLiveData<Int>()
-    val onFinishTimer = MutableLiveData<Boolean>()
+    // списки контактов/юзеров удовлетворящих поисковому запросу
+    val filteredContactsList = MutableLiveData<MutableList<Contact>>(mutableListOf())
+    val filteredUserList = MutableLiveData<MutableList<Contact>>(mutableListOf())
+
+    // наблюдаемые переменные, куда приходят результаты запроса к серверу
+    val registrationResult = MutableLiveData<Response<ServerResponse>?>()
+    val authorisationResult = MutableLiveData<Response<ServerResponse>?>()
+    val getUserContactsResult = MutableLiveData<Response<ContactsServerResponse>?>()
+    var getAllUsersResult = MutableLiveData<Response<ExtensionServerResponse>?>()
+    var addToContactListResult = MutableLiveData<Response<ContactsServerResponse>?>()
+    var deleteFromContactsResult = MutableLiveData<Response<ContactsServerResponse>?>()
+
+    val listWithAllContacts = MutableLiveData<List<Contact>>()                                      // список всех контактов пользователя
+    val listOfAllUsers = MutableLiveData<List<Contact>>()                                           // список ВСЕХ доступных для добавления юзеров
```

Comment:
Incapsulation violation. Also, not sure whether I would want to have nullable types here